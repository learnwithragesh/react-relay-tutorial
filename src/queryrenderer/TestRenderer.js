import React from 'react';
import graphql from 'babel-plugin-relay/macro';
import { QueryRenderer } from 'react-relay';

export default function TestRenderer({ environment, query }) {
    return (
        <QueryRenderer
            environment={environment}
            query={query}
            variables={{}}
            render={({error, props}) => {
                if (props) {
                    return <p>hh</p>;
                } else if (error) {
                    return error.message;
                }
                return 'Loading...';
            }}
        />
    );
}