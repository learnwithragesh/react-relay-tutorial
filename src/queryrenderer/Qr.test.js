/**
 * @jest-environment jsdom
 */
import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import {
  createMockEnvironment,
  MockPayloadGenerator,
} from 'relay-test-utils';
import graphql from 'babel-plugin-relay/macro';
import { QueryRenderer } from 'react-relay';
import "core-js/stable";
import "regenerator-runtime/runtime";
import QR from './index';
import TestRenderer from './TestRenderer';

const query = graphql`
query QrQuery @relay_test_operation {
  user(login: "") {
      id
  }
}
`;

test('Loading State', () => {
    const environment = createMockEnvironment();
    const renderer = ReactTestRenderer.create(
        <QR environment={environment}/>
    );

    console.log(renderer.toJSON());

    expect(renderer.toJSON().type).toBe('p');
});

test('Data Render', () => {
    const environment = createMockEnvironment();
    const renderer = ReactTestRenderer.create(
        <TestRenderer environment={environment} query={query}/>
    );
    ReactTestRenderer.act(() => {
        environment.mock.resolveMostRecentOperation(operation =>
          MockPayloadGenerator.generate(operation),
        );
    });
    console.log(renderer.toJSON());

    expect(renderer.toJSON().type).toBe('p');
  });