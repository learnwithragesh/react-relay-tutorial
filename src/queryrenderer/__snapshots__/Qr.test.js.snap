// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Data Render 1`] = `
<p>
  hh
</p>
`;

exports[`ItineraryCard 1`] = `
<p>
  hh
</p>
`;
